### PR TITLE
chore: add Flake8 test

### PR DESCRIPTION
**Pull request by Panaetius**
Adds flake8 tests since Black doesn't cover everything.

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -324,6 +324,9 @@ def request_callback(request):
         rsps.add_callback(responses.GET, "http://example.com/file", callback=request_callback)
         rsps.add_callback(responses.GET, "https://example.com/file", callback=request_callback)
         rsps.add_callback(responses.GET, "http://example.com/file.ext?foo=bar", callback=request_callback)
+        rsps.add_callback(responses.HEAD, "http://example.com/file", callback=request_callback)
+        rsps.add_callback(responses.HEAD, "https://example.com/file", callback=request_callback)
+        rsps.add_callback(responses.HEAD, "http://example.com/file.ext?foo=bar", callback=request_callback)
```
Comment (mohammad-sdsc) [inline conftest.py:329]: Is this required for this PR or is left from somewhere else?

File: renku/core/commands/dataset.py [hunk 2]
```diff
@@ -482,6 +482,11 @@ def import_dataset(
     client, uri, name="", extract=False, with_prompt=False, yes=False, commit_message=None, progress=None,
 ):
     """Import data from a 3rd party provider or another renku project."""
+    u = urllib.parse.urlparse(uri)
+    if u.scheme not in ("", "file", "git+https", "git+ssh"):
+        # NOTE: Check if the url is a redirect.
+        uri = requests.head(uri, allow_redirects=True).url
+
```
Comment (mohammad-sdsc) [inline renku/core/commands/dataset.py:489]: Same as above.

File: renku/core/management/datasets.py [hunk 3]
```diff
@@ -251,7 +251,7 @@ def add_data_to_dataset(
             )
         else:
             for url in urls:
-                is_remote, is_git = _check_url(url)
+                is_remote, is_git, url = _check_url(url)
```
Comment (mohammad-sdsc) [inline renku/core/management/datasets.py:254]: Same as above for the whole file.

File: renku/service/views/datasets.py [hunk 4]
```diff
@@ -29,8 +29,6 @@
     dataset_remove,
     edit_dataset,
     file_unlink,
-    list_datasets,
-    list_files,
```
Comment (mohammad-sdsc) [inline renku/service/views/datasets.py:33]: This requires a re-formatting.

**Review by mohammad-sdsc — COMMENTED** [review]
I guess this include codes from another PR. Looks good other than that!

File: run-tests.sh [hunk 5]
```diff
@@ -39,6 +39,7 @@ check_styles(){
     pydocstyle renku tests conftest.py docs
     black --check --diff renku tests conftest.py
     isort -c --df .
+    flake8 renku tests
```
Comment (mohammad-sdsc) [inline run-tests.sh:42]: We should include `conftest.py` and `setup.py` too.

File: conftest.py (hunk 1 above)
Comment (Panaetius) [reply conftest.py:329]: Weird that that ended up here. It's part of https://github.com/SwissDataScienceCenter/renku-python/pull/1516 I'll remove it here

**Review by mohammad-sdsc — APPROVED** [review]
Thanks! Looks good!

**Pull request merged**